README.md

```diff
@@ -283,15 +284,55 @@ enum Flags {
 
     // DataForAgentFlags is a bit mask. Values below define individual bits.
 
-    // The server asks the agent to report effective config.
+    // The server asks the agent to report effective config. This bit MUST NOT be
+    // set if the Agent indicated it cannot report effective config by setting
+    // the ReportsEffectiveConfig bit to 0 in StatusReport.capabilities field.
     ReportEffectiveConfig = 0x00000001;
 
-    // The server asks the agent to report addon statuses.
+    // The server asks the agent to report addon statuses. This bit MUST NOT be
+    // set if the Agent indicated it cannot report addon status by setting
+    // the ReportsAddonStatus bit to 0 in StatusReport.capabilities field.
     ReportAddonStatus     = 0x00000002;
 }
 ```
 
 
+#### capabilities
+
+Bitmask of flags defined by Capabilities enum. All bits that are not defined in
+Capabilities enum MUST be set to 0 by the Server. This allows extending the
+protocol and the Capabilities enum in the future such that old Servers
+automatically report that they don't support the new capability. This field MUST
+be set in the first DataForAgent sent by the Server and MAY be omitted in
+subsequent DataForAgent messages by setting it to Unspecified value.
+
+```protobuf
+enum Capabilities {
```

Comment:
Should we differentiate the two enums?
```suggestion
enum ServerCapabilities {
```

Reply:
They are differentiated by being nested in the corresponding message. But I like your suggestion, it makes things more clear.